fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/ReduceCastRule.java

```diff
@@ -70,6 +72,16 @@ public ScalarOperator visitCastOperator(CastOperator operator, ScalarOperatorRew
             return operator.getChild(0);
         }
 
+        if (operator.getChild(0).getType().isBoolean() && !operator.getChild(0).isNullable()
+                && !operator.getType().isBoolean()) {
```

Comment:
a question, the `operator.getChild(0).isNullable()` isn't credible....
like `select cast(b.xx) from (select * from a outer join b)`